Add configurable restore permission for restoreAudit action

zafermesut opened the pull request:
This PR adds a configurable restore permission for the restoreAudit action.
By default, the permission used is **restoreAudit**, but this can now be overridden using the plugin's _permission()_ method.
This allows projects with custom policies (e.g. restore_module) to integrate without modifying core package logic.

aSeriousDeveloper commented on src/FilamentAuditingPlugin.php:
```diff
@@ -49,6 +51,18 @@ public static function make(): static
         return app(static::class);
     }
 
+    public function permission(string $permission): static
+    {
+        $this->restorePermission = $permission;
+
+        return $this;
+    }
+
+    public function getRestorePermission(): string
+    {
+        return $this->restorePermission;
+    }
+
```
Would you be able to refactor this into its own trait, same as the other features?

aSeriousDeveloper commented on src/FilamentAuditingPlugin.php:
```diff
@@ -27,6 +27,8 @@ class FilamentAuditingPlugin implements Plugin
     use HasUsers;
     use HasUserSchema;
 
+    protected string $restorePermission = 'restoreAudit';
+
```
Would you be able to refactor this into its own trait, same as the other features?

aSeriousDeveloper reviewed (changes requested):
Good idea thanks.

However, please make sure the new feature in the `FilamentAuditingPlugin` is instead placed as a concern like the other configurable features.

Additionally, make sure it makes proper use of Closures.

zafermesut commented:
Thank you for your feedback!

I realized I had initially overlooked using a trait like the other configurable features. I’ve now refactored the permission functionality into a dedicated concern to better align with the overall structure.

I also added some tests to cover different permission scenarios. However, I'm not very experienced with writing tests yet, so I'm not entirely sure if they are sufficient or written in the best way.

If you get a chance, I’d really appreciate it if you could take a look and share any feedback. It would help me improve and learn from it.

aSeriousDeveloper commented:
Hi @zafermesut, thanks for the additional changes.
I decided to make some additional refactors of my own, you can check my 2 commits to see them.

It's mostly just moving around the code and the tests to get things a bit more organised on my end, but some of the main bits:
- Moved the new trait to be alongside the other top-level plugin traits
- Created a new action for restoration, with functions for checking auth & ability to restore
- rejigged your tests and merged them in with a new test class to go alongside the new action
- reworked permissions in my testing suite & added policies for them
